### PR TITLE
🧪 Bring over StackedNotificationsBehavior from Toolkit

### PR DESCRIPTION
**Pull request by michael-hawker**
Importing from Toolkit

See Details in #210

Been using as my test/demo for presentation, so been working alright. Want to get initial version in.

Made some additional small tweaks:
- Make links in Markdown clickable
- Remove close buttons from Tabs in experiment mode
- Remove empty data at end of docs (think was causing the rogue 'FrontMatter' to showup)
- tweak template to add blank line needed to render list in documentation
- rev packages to check that working with all updates

### CODE REVIEW

**Comment by michael-hawker** [comment]
Bubbling up the CI Uno error here:
```
"D:\a\Labs-Windows\Labs-Windows\Toolkit.Labs.All.sln" (default target) (1:2) ->
"D:\a\Labs-Windows\Labs-Windows\platforms\CommunityToolkit.Labs.Droid\CommunityToolkit.Labs.Droid.csproj" (default target) (6:6) ->
"D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj" (default target) (21:73) ->
(_UnoSourceGenerator target) -> 
  MSBUILD : error : Generation failed: System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\obj\Release\monoandroid11.0\g\NativeCtorsGenerator\StackedNotificationsBehaviorExperiment_Samples_StackedNotificationsBehaviorCustomSample_3aca12002872ed9de2d15c6b3ebbd506.g.cs'. [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.File.InternalWriteAllText(String path, String contents, Encoding encoding, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at Uno.SourceGeneration.Host.SourceGeneratorEngine.Generate() in D:\a\1\s\src\Uno.SourceGeneration.Engine.Shared\SourceGeneratorEngine.cs:line 285 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at Uno.SourceGeneration.Host.Program.RunGeneration(String generatedFilesOutputPath, String binlogOutputPath, BuildEnvironment environment, Boolean enableConsole) in D:\a\1\s\src\Uno.SourceGeneration.Host\Program.SingleUse.cs:line 68 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
  C:\Users\runneradmin\.nuget\packages\uno.sourcegenerationtasks\4.0.2\build\netstandard1.0\Uno.SourceGenerationTasks.targets(134,2): error : Generation failed, error code 3 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
"D:\a\Labs-Windows\Labs-Windows\Toolkit.Labs.All.sln" (default target) (1:2) ->
"D:\a\Labs-Windows\Labs-Windows\platforms\CommunityToolkit.Labs.iOS\CommunityToolkit.Labs.iOS.csproj" (default target) (7:6) ->
"D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj" (default target) (21:75) ->
  MSBUILD : error : Generation failed: System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\obj\Release\xamarinios10\g\NativeCtorsGenerator\StackedNotificationsBehaviorExperiment_Samples_StackedNotificationsBehaviorCustomSample_3aca12002872ed9de2d15c6b3ebbd506.g.cs'. [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.File.InternalWriteAllText(String path, String contents, Encoding encoding, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at Uno.SourceGeneration.Host.SourceGeneratorEngine.Generate() in D:\a\1\s\src\Uno.SourceGeneration.Engine.Shared\SourceGeneratorEngine.cs:line 285 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at Uno.SourceGeneration.Host.Program.RunGeneration(String generatedFilesOutputPath, String binlogOutputPath, BuildEnvironment environment, Boolean enableConsole) in D:\a\1\s\src\Uno.SourceGeneration.Host\Program.SingleUse.cs:line 68 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
  C:\Users\runneradmin\.nuget\packages\uno.sourcegenerationtasks\4.0.2\build\netstandard1.0\Uno.SourceGenerationTasks.targets(134,2): error : Generation failed, error code 3 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
"D:\a\Labs-Windows\Labs-Windows\Toolkit.Labs.All.sln" (default target) (1:2) ->
"D:\a\Labs-Windows\Labs-Windows\platforms\CommunityToolkit.Labs.macOS\CommunityToolkit.Labs.macOS.csproj" (default target) (8:6) ->
"D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj" (default target) (21:77) ->
  MSBUILD : error : Generation failed: System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\obj\Release\xamarinmac20\g\NativeCtorsGenerator\StackedNotificationsBehaviorExperiment_Samples_StackedNotificationsBehaviorCustomSample_3aca12002872ed9de2d15c6b3ebbd506.g.cs'. [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at System.IO.File.InternalWriteAllText(String path, String contents, Encoding encoding, Boolean checkHost) [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at Uno.SourceGeneration.Host.SourceGeneratorEngine.Generate() in D:\a\1\s\src\Uno.SourceGeneration.Engine.Shared\SourceGeneratorEngine.cs:line 285 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
MSBUILD : error :    at Uno.SourceGeneration.Host.Program.RunGeneration(String generatedFilesOutputPath, String binlogOutputPath, BuildEnvironment environment, Boolean enableConsole) in D:\a\1\s\src\Uno.SourceGeneration.Host\Program.SingleUse.cs:line 68 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
  C:\Users\runneradmin\.nuget\packages\uno.sourcegenerationtasks\4.0.2\build\netstandard1.0\Uno.SourceGenerationTasks.targets(134,2): error : Generation failed, error code 3 [D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehavior.Samples.csproj]
```

Seems internal, but also going to rebase the branch and try rebuilding now that we fixed the SG test issue.

**Comment by michael-hawker** [comment]
Rebased, we'll see what happens now.

**Comment by michael-hawker** [comment]
Hmm, just checked the path specified here: `D:\a\Labs-Windows\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\obj\Release\xamarinios10\g\NativeCtorsGenerator\StackedNotificationsBehaviorExperiment_Samples_StackedNotificationsBehaviorCustomSample_3aca12002872ed9de2d15c6b3ebbd506.g.cs` is **283** characters. I think the regular path limit is 260? So we might have to configure this in the CI?

**Comment by michael-hawker** [comment]
Uno folks pointed us to this issue: https://github.com/dotnet/msbuild/issues/53#issuecomment-1175850826

I'm wondering if we can set the Reg key for the VM in powershell and run the git command as we did locally in our workflow, from our template doc here: https://github.com/CommunityToolkit/Labs-Windows/tree/main/template#long-paths

@Arlodotexe maybe something to try quick and see what happens? Otherwise, we can try and sort out later when we sync with the Uno folks in August.

**Comment by Arlodotexe** [comment]
> I'm wondering if we can set the Reg key for the VM in powershell and run the git command as we did locally in our workflow, from our template doc here: https://github.com/CommunityToolkit/Labs-Windows/tree/main/template#long-paths

In addition to the Git command, it looks like there's a PowerShell command for this too. We should be able to use a similar command to _read_ the value and see if it's enabled already.

```powershell
Set-ItemProperty 'HKLM:\System\CurrentControlSet\Control\FileSystem' -Name 'LongPathsEnabled' -value 1
```

There's 2 considerations to hash out still:
- It takes a full restart to take effect. We'd have to interrupt the user's workflow completely for a restart if they don't already have this enabled.
- Where would we run this script from? Do we try to integrate the check into MSBuild?

**Comment by Arlodotexe** [comment]
Before we can merge this, we need to manually add in changes done for #226

**Comment by michael-hawker** [comment]
Well at least we're getting another error now? (And it worked on the WinUI 3 build!)

Seems .NET Native related now:
```
"D:\a\Labs-Windows\Labs-Windows\Toolkit.Labs.All.sln" (default target) (1:2) ->
"D:\a\Labs-Windows\Labs-Windows\platforms\CommunityToolkit.Labs.Uwp\CommunityToolkit.Labs.Uwp.csproj" (default target) (12:6) ->
(BuildNativePackage target) -> 
  C:\Users\runneradmin\.nuget\packages\microsoft.net.native.compiler\2.2.10-rel-29722-00\tools\Microsoft.NetNative.targets(805,5): error : ILT0005: 'C:\Users\runneradmin\.nuget\packages\runtime.win10-x64.microsoft.net.native.compiler\2.2.10-rel-29722-00\tools\x64\ilc\Tools\nutc_driver.exe @"D:\a\Labs-Windows\Labs-Windows\platforms\CommunityToolkit.Labs.Uwp\obj\x64\Release\ilc\intermediate\MDIL\CommunityToolkit.Labs.Uwp.rsp"' returned exit code 1 [D:\a\Labs-Windows\Labs-Windows\platforms\CommunityToolkit.Labs.Uwp\CommunityToolkit.Labs.Uwp.csproj]
```

Going to try the fix suggested here and see what happens: https://dev.to/mhmd_azeez/solving-nutcdriver-exe-returned-code-1-error-while-building-uwp-app-in-release-mode-2mc6

(And the test space changes seem to work, think there may be more we can do hear in the future, but them being connected in the GitHub commit check is the first step.)

**Comment by Arlodotexe** [comment]
@michael-hawker See #245.

File: common/Labs.Head.Uwp.props [hunk 1]
```diff
@@ -9,4 +9,8 @@
   <PropertyGroup>
     <TargetFramework>$(UwpTargetFramework)</TargetFramework>
   </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)' == 'Release'">
+    <UseDotNetNativeToolchain>true</UseDotNetNativeToolchain>
+  </PropertyGroup>
```
Comment (michael-hawker) [inline common/Labs.Head.Uwp.props:15]: This should remain in the Uwp.Base.props file for the test projects being built in release mode as well.

**Comment by michael-hawker** [comment]
@Arlodotexe think we discovered something. Reran the failed job, but appears that there's something in the test projects (maybe related to XML) that's tripping up the .NET Native compiler more consistently?

When we abstracted running the tests from the .NET Native compiler, then things were fine.

Maybe we do go back and not run tests through .NET Native, and add smoke tests later that use .NET Native...? 🤔

FYI @tommcdon, have you heard or aware of any of these issues with MSTest and .NET Native?

**Comment by Arlodotexe** [comment]
This branch is now on par with `main` in terms of the ILT0005 error. We should be good to merge.

I've added details in #245, let's continue the discussion there

**Pull request merged**